shell/platform/linux/fl_backing_store_provider.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "fl_backing_store_provider.h"
+
+#include <epoxy/gl.h>
+
+struct _FlBackingStoreProvider {
+  GObject parent_instance;
+
+  uint32_t framebuffer_id;
+  uint32_t texture_id;
+  GdkRectangle geometry;
+};
+
+G_DEFINE_TYPE(FlBackingStoreProvider, fl_backing_store_provider, G_TYPE_OBJECT)
+
+static void fl_backing_store_provider_dispose(GObject* object) {
+  FlBackingStoreProvider* self = FL_BACKING_STORE_PROVIDER(object);
+
+  glDeleteFramebuffers(1, &self->framebuffer_id);
+  glDeleteTextures(1, &self->texture_id);
+
+  G_OBJECT_CLASS(fl_backing_store_provider_parent_class)->dispose(object);
+}
+
+static void fl_backing_store_provider_class_init(
+    FlBackingStoreProviderClass* klass) {
+  G_OBJECT_CLASS(klass)->dispose = fl_backing_store_provider_dispose;
+}
+
+static void fl_backing_store_provider_init(FlBackingStoreProvider* self) {}
+
+FlBackingStoreProvider* fl_backing_store_provider_new(int width, int height) {
+  FlBackingStoreProvider* provider = FL_BACKING_STORE_PROVIDER(
+      g_object_new(fl_backing_store_provider_get_type(), nullptr));
+
+  glGenTextures(1, &provider->texture_id);
+  glGenFramebuffers(1, &provider->framebuffer_id);
+
+  glBindFramebuffer(GL_FRAMEBUFFER, provider->framebuffer_id);
```

Comment:
Binding framebuffer 0 does not mean 'unbinding' framebuffer, but binding to the system default framebuffer. Flutter/Skia will always bind a specific framebuffer (provided by `FlBackingStoreProvider`) before rendering. So we have no need to bind to framebuffer 0 here. Restoring framebuffer binding requires a call to `glGetIntegerv(GL_FRAMEBUFFER_BINDING, &old_fbo);`.

macOS implementation has the same behavior, and because framebuffer 0 is not the default framebuffer.
https://github.com/flutter/engine/blob/0bf71f61401bbe48bc01636ab1a98f447ca57633/shell/platform/darwin/macos/framework/Source/FlutterFrameBufferProvider.mm#L31-L37